tests/testCentroidChecker.py

```diff
@@ -0,0 +1,171 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import numpy
+import unittest
+
+import lsst.utils.tests
+import lsst.meas.base
+import lsst.meas.base.tests
+import lsst.meas.algorithms
+import lsst.afw.table
+from lsst.meas.base.tests import (AlgorithmTestCase)
+from lsst.meas.base.flagDecorator import addFlagHandler
+import lsst.pex.exceptions
+from lsst.meas.base.pluginRegistry import register
+from lsst.meas.base.sfm import SingleFramePluginConfig, SingleFramePlugin
+
+
+class CentroiderConfig(SingleFramePluginConfig):
+
+    moveX = lsst.pex.config.Field(dtype=int, default=0, optional=False,
+                                  doc="amount to re-position in X")
+    moveY = lsst.pex.config.Field(dtype=int, default=0, optional=False,
+                                  doc="amount to re-position in Y")
+    dist = lsst.pex.config.Field(dtype=int, default=None, optional=False,
+                                 doc="distance to allow centroid to be off")
+
+
+@register("test_Centroider")
+@addFlagHandler(("flag", "General Failure error"), ("test_flag", "second flag"))
+class Centroider(SingleFramePlugin):
+    '''
+    This is a sample Python plugin.  The flag handler for this plugin is created
+    during construction, and is called using the method fail().  All plugins are
+    required to implement this method, which is used to set the flags in the
+    output source record if an error occurs.
+    '''
+    ConfigClass = CentroiderConfig
+    # Class variables ErrEnum and FLAGDEFS are added by the decorator
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.CENTROID_ORDER
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+
+        self.xKey = schema.addField(schema.join(name, "x"), type=numpy.float64)
+        self.yKey = schema.addField(schema.join(name, "y"), type=numpy.float64)
+        if self.config.dist is None:
+            self.centroidChecker = lsst.meas.base.CentroidChecker(schema, name)
+        else:
+            self.centroidChecker = lsst.meas.base.CentroidChecker(schema, name, self.config.dist)
+
+    #   This is a measure routine which does nothing except to raise Exceptions
+    #   as requested by the caller. Errors normally don't occur unless there is
+    #   something wrong in the inputs, or if there is an error during the measurement
```

Comment:
I find "as requested by the caller" a bit confusing here.  do you mean "Exceptions caused by moving the true result the amount specified in the config"?